static/app/views/performance/landing/widgets/transforms/transformMetricsToArea.tsx

```diff
@@ -104,28 +117,40 @@ export function transformMetricsToArea<T extends WidgetDataConstraint>(
   const previousTotalPerBucket = isFailureRateWidget
     ? responsePrevious?.intervals.map((_intervalValue, intervalIndex) =>
         responsePrevious?.groups.reduce(
-          (acc, group) => acc + group.series[metricsField][intervalIndex],
+          (acc, group) => acc + (group.series[metricsField][intervalIndex] ?? 0),
           0
         )
       )
     : undefined;
 
-  const previousData = previousGroups?.map(group => ({
-    seriesName: `previous ${metricsField}`,
-    data: response?.intervals.map((intervalValue, intervalIndex) => ({
-      name: moment(intervalValue).valueOf(),
-      value: defined(previousTotalPerBucket)
-        ? group.series[metricsField][intervalIndex] /
-          previousTotalPerBucket[intervalIndex]
-        : group.series[metricsField][intervalIndex],
-    })),
-    stack: 'previous',
-  }));
+  const previousData = previousGroups?.map(group => {
+    const series = response?.intervals.map((intervalValue, intervalIndex) => {
+      const serieBucket = group.series[metricsField][intervalIndex];
+      const totalSerieBucket = previousTotalPerBucket?.[intervalIndex];
+
+      return {
+        name: moment(intervalValue).valueOf(),
+        value:
+          defined(totalSerieBucket) &&
+          defined(serieBucket) &&
+          serieBucket > 0 &&
+          totalSerieBucket > 0
+            ? serieBucket / totalSerieBucket
+            : serieBucket,
+      };
+    });
+
+    return {
+      seriesName: `previous ${metricsField}`,
+      stack: 'previous',
+      data: series.some(serie => defined(serie.value)) ? series : [],
+    };
+  }) as null | Series[];
 
   return {
     ...commonChildData,
     hasData: defined(data) && !!data.length && !!data[0].data.length,
-    data,
+    data: data as Series[],
```

Comment:
the tricky part is the `value: number | null` and the correct thing to do is to add the `null` type in the https://github.com/getsentry/sentry/blob/20f152a39e3e1160fbb598c8da10c9eb37874b7f/static/app/types/echarts.tsx#L5 instead of casting....but I would have to update so many different parts of the application, something I prefer to not do now as it can be risky